pkg/serverless/trigger/events.go

```diff
@@ -167,7 +174,7 @@ func Unmarshal(payload []byte) (map[string]any, error) {
 }
 
 func isAPIGatewayEvent(event map[string]any) bool {
-	return json.GetNestedValue(event, "requestcontext") != nil &&
+	return json.GetNestedValue(event, "requestcontext", "stage") != nil &&
```

Comment:
Is this change necessary? I'm struggling to understand why we need to make this check more restrictive.

Reply:
it's a revert of the change we did in the previous release in https://github.com/DataDog/datadog-agent/pull/19569 to restore it the way it was